breakpad-symbols/src/http.rs

```diff
@@ -470,8 +538,43 @@ impl SymbolSupplier for HttpSymbolSupplier {
         &self,
         module: &(dyn Module + Sync),
     ) -> Result<SymbolFile, SymbolError> {
+        // If we don't have a debug_file or debug_identifier, then try to get it
+        // from a symbol server.
+        let mut debug_file = module.debug_file().unwrap_or_default().into_owned();
+        let mut debug_id = module.debug_identifier().unwrap_or_default();
+
+        if debug_file.to_string().is_empty() || debug_id.is_nil() {
+            debug!("Missing debug file or debug identifier--trying lookup with code info");
+            match lookup_debug_info_by_code_info(&self.urls, module).await {
+                Some(debug_info_result) => {
+                    debug_file = debug_info_result.debug_file;
+                    debug_id = debug_info_result.debug_identifier;
+                },
+                _ => (),
+            }
+        }
```

Comment:
In order for the lookup to help with the cache, local files, and symbol servers, we have to do the lookup and then carry the new debug file and debug id through all the other functions. So I did the lookup first if the debug file or debug identifier is empty. Then I carried the (possibly new) debug file and debug identifier through all the other functions by putting the needed data in a `SimpleModule`. It was kind of painful because `SimpleModule` isn't like minidump `Module`.

If you have ideas on how to do this better, I'm all ears.